src/filters/ws.rs

```diff
@@ -260,6 +256,10 @@ impl Stream for WebSocket {
                 protocol::Message::Pong(payload) => {
                     trace!("websocket client pong: {:?}", payload);
                 }
+                protocol::Message::Close(frame) => {
+                    trace!("websocket closed: {:?}", frame);
+                    return Ok(Async::Ready(None));
```

Comment:
I'm not sure that it's correct to return `None` here. `Message::Close` means that we received a `Close` message, so I expect that the user received the `Close` message back (because basically that's what happened).

I think the more correct way to handle it would be to impement it the same way it's done in `tokio-tungstenite` : https://github.com/snapview/tokio-tungstenite/commit/e2cdf11bc8d8d0837d2073ecc3db686c13435523 , i.e.

`Message::Close` -> `Ready(Some(Message::Close))`
`ConnectionClosed` -> `Ready(None)`

Reply:
Well I don't think I changed anything behavior-wise. The `Err(ConnectionClosed)` was changed to `Ok(Message::Closed)` AFAICT, this is just updating a dependency, not trying to fix any behavior. Unless I'm missing something, it should be another issue / MR to fix that.

Reply:
The `Err(ConnectionClosed)` was not replaced by `Message::Closed`. We changed the semantics a bit. In the previous version, the `ConnectionClosed` was returned after the connection is closed and it contained an optional close code.

Since there were requests to support sending/receiving `Closed` as a message (and also it was required for `tokio-tungstenite` to make it compatible with the "expected semantics" of `Stream`), we decided to make `Closed` as a part of the `Message`.

Because of that, the `ConnectionClosed` does not contain an optional close code in it anymore, but it still can be returned from (see https://github.com/snapview/tungstenite-rs/commit/b8f7d3668ec79c62abb9562ee3b5dd355908b6a5).

Reply:
Oh ok, my bad then. Thanks for the explanation. I'll fix it tonight when I get home